ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestUtils.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.apitesting
+
+import java.util.UUID
+
+import com.digitalasset.daml.lf.archive.UniversalArchiveReader
+import com.digitalasset.ledger.api.v1.command_service.SubmitAndWaitRequest
+import com.digitalasset.ledger.api.v1.command_submission_service.SubmitRequest
+import com.digitalasset.ledger.api.v1.commands.Command.Command.Create
+import com.digitalasset.ledger.api.v1.commands.{Command, Commands, CreateCommand}
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset.LedgerBoundary.{
+  LEDGER_BEGIN,
+  LEDGER_END
+}
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset.Value.Boundary
+import com.digitalasset.ledger.api.v1.transaction_filter.{
+  Filters,
+  InclusiveFilters,
+  TransactionFilter
+}
+import com.digitalasset.ledger.api.v1.value.Value.Sum.Party
+import com.digitalasset.ledger.api.v1.value.{Identifier, Record, RecordField, Value}
+import com.digitalasset.platform.PlatformApplications.Config
+import com.google.protobuf.timestamp.Timestamp
+import scalaz.syntax.tag._
+
+object TransactionFilters {
+  def allForParty(party: String) = TransactionFilter(Map(party -> Filters()))
+
+  def onlyTemplatesByParty(templatesByParty: Map[String, Seq[Identifier]]) =
+    TransactionFilter(
+      templatesByParty.mapValues(templateIds => Filters(Some(InclusiveFilters(templateIds)))))
+}
+
+object LedgerOffsets {
+  val LedgerBegin = LedgerOffset(Boundary(LEDGER_BEGIN))
+  val LedgerEnd = LedgerOffset(Boundary(LEDGER_END))
+}
+
+object TestParties {
+  val Alice = "Alice"
+  val Bob = "Bob"
+}
+
+object TestUtils {
+
+  private lazy val ParsedPackageId: String =
+    UniversalArchiveReader().readFile(Config.defaultDarFile).get.main._1 // Test.dar
+
+  val TestTemplateIds = TestTemplateIdentifiers(ParsedPackageId)
+
+}
+
+class TestUtils(
+    ledgerContext: LedgerContext,
+    applicationId: String = "test-application",
+    workflowId: String = "test-workflow") {
+
+  private val LedgerEffectiveTime = Timestamp(0L, 0)
+  private val MaximumRecordTime =
+    LedgerEffectiveTime.copy(seconds = LedgerEffectiveTime.seconds + 30L)
+
+  private def randomCommandId: String = UUID.randomUUID().toString
```

Comment:
This will have to be plugged into the machinery I am setting up to control if identifiers are randomized or not. My draft for this is here: https://github.com/digital-asset/daml/pull/1735/files#diff-5f0c4de564dbea036385a0cda87b7fd9R5

Reply:
ok, let me know when that lands. Note, that this part of the code is not used yet.